src/coreclr/gc/gc.cpp

```diff
@@ -44523,6 +44523,19 @@ void gc_heap::walk_heap_per_heap (walk_fn fn, void* context, int gen_number, BOO
                 end = heap_segment_allocated (seg);
                 continue;
             }
+#ifdef USE_REGIONS
+            else if (gen_number > 0)
+            {
+                // advance to next lower generation
+                gen_number--;
+                gen = gc_heap::generation_of (gen_number);
+                seg = generation_start_segment (gen);
```

Comment:
nit - you could write this as 

`seg = generation_start_segment (generation_of (gen_number));`

since it's very common to write it this way

Reply:
Yes, of course - but the initialization of these variables is happening at the top of the function in the style that I used, so I'm consistent with that...